src/components/pages/index-page/IndexPage.store.ts

```diff
@@ -0,0 +1,56 @@
+import { observable } from 'mobx';
+import { getCastsRequest, getShowRequest } from '../../../domains/shows/shows.services';
+import { initialResponseStatus } from '../../../utils/mobx.utils';
+import { ICast, IShow } from '../../../domains/shows/shows.types';
+import { ApiResponse } from '../../../utils/http/http.types';
+import { defaultShowId } from '../../../domains/shows/shows.constants';
+import orderBy from 'lodash.orderby';
+
+export const IndexPageStore = () =>
+  observable({
+    // globalStore: getGlobalStore(),
+    sortValue: '',
+    showResults: initialResponseStatus<IShow | null>(null),
+    castsResults: initialResponseStatus<ICast[]>([]),
+
+    get isRequesting(): boolean {
+      return [this.showResults.isRequesting, this.castsResults.isRequesting].some(Boolean);
+    },
+
+    get actors(): ICast[] {
+      return orderBy(this.castsResults.data, (cast) => cast.person[this.sortValue], 'asc');
+    },
+
+    setSortOption(sortValue: string): void {
+      this.sortValue = sortValue;
+    },
+
+    /**
+     * Store initializer. Should only be called once.
+     */
+    *init() {
+      yield Promise.all([this.loadShow(), this.loadCasts()]);
+    },
+
+    *loadShow() {
+      const response: ApiResponse<IShow> = yield getShowRequest(defaultShowId);
+
+      this.showResults = {
+        data: this.showResults.data,
+        isRequesting: false,
+        ...response, // Overwrites the default data prop or adds an error. Also adds the statusCode.
+      };
+    },
```

Comment:
@scottbromander

> One thing really stands out to me as odd. You use the show ID in the url params of the episode list, but not the index. 

I was thinking of having a dropdown with id's on the homepage to set the show id but I didn't want to do more than I needed for the example.

Reply:
@scottbromander what would you do?